packages/debug/src/browser/debug-session-connection.ts

```diff
@@ -215,7 +215,7 @@ export class DebugSessionConnection implements Disposable {
             };
 
             this.pendingRequests.set(request.seq, result);
-            if (timeout) {
+            if (typeof timeout === 'number') {
```

Comment:
> Unless there's a reason for this change, I would revert it.

OK, let's revert it. I do not understand if the entire debugger logic is based on VS Code and [Code uses `typeof timeout === 'number'` condition](https://github.com/microsoft/vscode/blob/1148f03a4c4aa18768d2dfae8fe0e8e8f7584ed7/src/vs/workbench/contrib/debug/common/abstractDebugAdapter.ts#L80), why Theia should use a different one.

Reply:
It's not based on VS Code, it's based on the the debug adapter protocol, but an independent implementation (otherwise, the copyright notice would indicate it). As a community, we're trying not to break adapters if there is no upside to it. I was simply asking what the upside is.

Reply:
https://github.com/eclipse-theia/theia/pull/11984/commits/e10d75eeeb447e99c785144b644f418d8578eeba